### ISSUE TITLE
`cast_nullable_to_non_nullable` to include null check of a nullable expression.

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#3905]
It seems when the user is encountered with [cast_nullable_to_non_nullable](https://dart-lang.github.io/linter/lints/cast_nullable_to_non_nullable.html)

```dart
void f(Map<String, Object?> json) {
  String s = json['a'] as String;
}
```

there is a tendency to write
```dart
String s = (json['a'] as String?)!;
```

instead of
```dart
String s = json['a']! as String;
```

Flutter has some instances (see https://github.com/flutter/flutter/pull/117026)

How about extending the lint to also report the case of ` (json['a'] as String?)!`?

**Comment by bwilkerson** [comment]
That code does look cleaner to me, but it isn't clear to me that `cast_nullable_to_non_nullable` should be extended to cover that case (because the code isn't casting to a non-nullable type).

I could potentially see this as a lint to promote applying the null check earlier, but I don't know how common this case is in code or even whether other people agree with my assessment that the code is cleaner. I think we should minimally wait to see what the response is to https://github.com/flutter/flutter/pull/117026.

**Comment by asashour** [comment]
This request was sent, because I assumed that `x as String` is the same as `x!`, in which both of them cast nullable `x` to a `String` for instance, but now I see the little difference, and both of them cannot be prevented.

The Flutter PR was merged, but yes, this could be a non-common case.
